services/app-web/src/testHelpers/apolloHelpers.tsx

```diff
@@ -836,25 +852,94 @@ const mockSubmittedHealthPlanPackageWithRevision = (): HealthPlanPackage => {
     }
 }
 
-type updateHealthPlanFormDataMockSuccessProps = {
+type updateHealthPlanFormDataMockProps = {
     pkg?: HealthPlanPackage
     updatedFormData: string
     id: string
+    statusCode?: 200 | 403 | 404 | 500
 }
 
-const updateHealthPlanFormDataMockSuccess = ({
+const updateHealthPlanFormDataMock = ({
     pkg = mockUnlockedHealthPlanPackage(),
     updatedFormData,
     id,
-}: updateHealthPlanFormDataMockSuccessProps): MockedResponse<UpdateHealthPlanFormDataMutation> => {
-    return {
-        request: {
-            query: UpdateHealthPlanFormDataDocument,
-            variables: {
-                input: { pkgID: id, healthPlanFormData: updatedFormData },
-            },
-        },
-        result: { data: { updateHealthPlanFormData: { pkg } } },
+    statusCode,
+}: updateHealthPlanFormDataMockProps): MockedResponse<UpdateHealthPlanFormDataMutation> => {
+    switch (statusCode) {
+        case 200:
+            return {
+                request: {
+                    query: UpdateHealthPlanFormDataDocument,
+                    variables: {
+                        input: {
+                            pkgID: id,
+                            healthPlanFormData: updatedFormData,
+                        },
+                    },
+                },
+                result: { data: { updateHealthPlanFormData: { pkg } } },
+            }
+        case 403:
+            return {
+                request: { query: UpdateHealthPlanFormDataDocument },
+                error: new Error('You are not logged in'),
+            }
+        case 404:
+            return {
+                request: {
+                    query: UpdateHealthPlanFormDataDocument,
+                    variables: {
+                        input: {
+                            pkgID: id,
+                            healthPlanFormData: updatedFormData,
+                        },
+                    },
+                },
+                result: { data: undefined },
```

Comment:
I don’t think this is what is returned on a 404 for UpdateHPFD. 404 returns something that looks like this for the fetch query:

```
{
    "data": {
        "fetchHealthPlanPackage": {
            "pkg": null,
            "__typename": "FetchHealthPlanPackagePayload"
        }
    }
}
```

But, while fetching a non existant package is treated as not an error, and we just return undefined, attempting to update a package that does not exist is treated as an error:

from the resolver:
```
            throw new UserInputError(errMessage, {
                argumentName: 'pgkID',
            })
```